utils/p4runtime_lib/convert.py

```diff
@@ -42,6 +42,16 @@ def encodeIPv4(ip_addr_string):
 
 def decodeIPv4(encoded_ip_addr):
     return socket.inet_ntoa(encoded_ip_addr)
+    
+ipv6_pattern = re.compile('([a-f0-9:]+:+)+[a-f0-9]+')
+def matchesIPv6(ip_addr_string):
+    return ipv6_pattern.match(ip_addr_string) is not None
```

Comment:
There are other forms of IPv6 addresses that are legal syntax.  I do not know which built-in Python function would recognize them all off the top of my head, but I would recommend searching for one.  Perhaps it might be `socket.inet_pton` with a try/catch block around it?

Reply:
Great suggestion! , implemented required changes.